hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/thrift/HMSClient.java

```diff
@@ -0,0 +1,456 @@
+                               @NotNull List<String> arguments)
+      throws TException {
+    return client.drop_partition(dbName, tableName, arguments, true);
+  }
+
+  public List<Partition> getPartitions(@NotNull String dbName, @NotNull String tableName) throws TException {
+    return client.get_partitions(dbName, tableName, (short) -1);
+  }
+
+  public DropPartitionsResult dropPartitions(@NotNull String dbName, @NotNull String tableName,
+                                             @Nullable List<String> partNames) throws TException {
+    if (partNames == null) {
+      return dropPartitions(dbName, tableName, getPartitionNames(dbName, tableName));
+    }
+    if (partNames.isEmpty()) {
+      return null;
+    }
+    return client.drop_partitions_req(new DropPartitionsRequest(dbName,
+        tableName, RequestPartsSpec.names(partNames)));
+  }
+
+  public List<Partition> getPartitionsByNames(@NotNull String dbName, @NotNull String tableName,
+                                              @Nullable List<String> names) throws TException {
+    if (names == null) {
+      return client.get_partitions_by_names(dbName, tableName,
+          getPartitionNames(dbName, tableName));
+    }
+    return client.get_partitions_by_names(dbName, tableName, names);
+  }
+
+  public boolean alterTable(@NotNull String dbName, @NotNull String tableName, @NotNull Table newTable)
+      throws TException {
+    client.alter_table(dbName, tableName, newTable);
+    return true;
+  }
+
+  public void alterPartition(@NotNull String dbName, @NotNull String tableName,
+                             @NotNull Partition partition) throws TException {
+    client.alter_partition(dbName, tableName, partition);
+  }
+
+  public void alterPartitions(@NotNull String dbName, @NotNull String tableName,
+                              @NotNull List<Partition> partitions) throws TException {
+    client.alter_partitions(dbName, tableName, partitions);
+  }
+
+  public void appendPartition(@NotNull String dbName, @NotNull String tableName,
+                              @NotNull List<String> partitionValues) throws TException {
+    client.append_partition_with_environment_context(dbName, tableName, partitionValues, null);
+  }
+
+  private TTransport open(HiveConf conf, @NotNull URI uri) throws
+      TException, IOException, LoginException {
+    boolean useSasl = conf.getBoolVar(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL);
+    boolean useFramedTransport = conf.getBoolVar(HiveConf.ConfVars.METASTORE_USE_THRIFT_FRAMED_TRANSPORT);
+    boolean useCompactProtocol = conf.getBoolVar(HiveConf.ConfVars.METASTORE_USE_THRIFT_COMPACT_PROTOCOL);
+    LOG.debug("Connecting to {}, framedTransport = {}", uri, useFramedTransport);
+
+    transport = new TSocket(uri.getHost(), uri.getPort(), (int) SOCKET_TIMEOUT_MS);
+
+    if (useSasl) {
+      LOG.debug("Using SASL authentication");
+      HadoopThriftAuthBridge.Client authBridge =
```

Comment:
I don't have good knowledge on these. But I assume you have tested this and hence gonna skim through.